### ISSUE TITLE
mzML profile mode?

### ISSUE THREAD

**Issue by rmflight** [compomics/ThermoRawFileParser#34]
From what I can tell, this tool only outputs centroided data, even when the data is acquired in profile mode. 

It would be nice to have the option to output either centroided or profile data when samples are acquired in profile mode. Very useful for some metabolomics applications.

**Comment by nielshulstaert** [comment]
Hi,

Thanks for your input. I decided to output the centroided data when available after a mail conversation with a very helpful expert at thermo. It's important to note that not all profile scans have centroided data available and we don't do any post-processing besides the native thermo library. That's also why we have the "MGF with profile data excluded" option. 

I agree that the user should be able to choose, is it ok if I add a flag and keep the default behaviour?

**Comment by rmflight** [comment]
I think it's great if the default is to output the Thermo peak centroided data. However, that should be documented, and having a flag to get the profile data instead would be fine.

**Comment by ypriverol** [comment]
@rmflight can you test the new version. We have address the following issue.

**Comment by timosachsenberg** [comment]
I am pretty sure I did not change anything in our call to the TRFP but the spectrum does not get picked anymore.

**Comment by rmflight** [comment]
For v112 vs v114, I get file sizes of 1749854 and 1749796, respectively when I specify `-f 1`, and don't do anything for peak picking. 

When I add the `-p` option in v 1.1.4, my file size jumps to 23974569, or 13X the size of the other two. So from that simple perspective, the `-p` is definitely reporting profile mode spectra. I will read them into R using `xcms` hopefully on Friday (currently on vacation), but wanted to download and get this checked ASAP.

**Comment by timosachsenberg** [comment]
@rmflight does this work for you with the v115 version and using the new outfile= option?
We currently have the strange situation that with peak picking enabled (noPeakPicking flag is not set) we get profile data:
https://github.com/OpenMS/OpenMS/pull/4095
Can anybody confirm that?

**Comment by jpfeuffer** [comment]
At least they are declared as profile spectra according to the CV term that is created here (without considering the new flag somehow):
https://github.com/compomics/ThermoRawFileParser/blob/a4038c3499435b3aec44b600ef7dc5f0a6e05550/Writer/MzMlSpectrumWriter.cs#L951

**Comment by rmflight** [comment]
Won't be able to check until tomorrow, unfortunately. Currently on
vacation, and driving all day today. Should be able to check tomorrow.

**Comment by rmflight** [comment]
OK, so both v114 and v115 have `profile` in the output type, there is no `centroid` to be found, even when thermo peak picking is asked for. This is in there even when I specify the output file directly using `-b`. So even though peak picking is being performed, the output does not reflect that. I'm guessing there is a small bug, as pointed out by @jpfeuffer

**Comment by nielshulstaert** [comment]
Hi if you run the parser without the `-p` flag, the native thermo library centroiding is applied. In the mzML output format, both centroid en profile scans are reported.

Using the `-p` flag disables the thermo centroiding and results in bigger files. In the mzML output format, both centroid en profile scans are reported. 

So using the `-p` flag results in the same number of spectra as not using it. It's like not selecting the vendor peak picking in msconvert.

Could you provide a link with the RAW file for which you don't get centroided scans?

**Comment by timosachsenberg** [comment]
Same file. As Julianus linked in this issue We think just the metadata is not properly updated but centroiding is performed.

**Comment by rmflight** [comment]
Here is a file I've been using, in case it helps to have another one: https://drive.google.com/open?id=1AWSEQV7TtnFtPHwzld5U1KWtz1aM6kUm

**Comment by nielshulstaert** [comment]
Hi @timosachsenberg what do you mean with the metadata?

**Comment by jpfeuffer** [comment]
Hey! I think he means the cvparams preceding the spectrum:
We always get
`<cvParam cvRef="MS" accession="MS:1000128" value="" name="profile spectrum" />`
now, although peak picking is activated.

**Comment by rmflight** [comment]
Right. If you do a search for `centroid`, it is not there, even if peak picking is activated. Previously, in v112, the metadata would contain `centroided spectrum` (or something similar), indicating that the spectrum was peak picked. Now, it never does, whether peak picking is activated or not.

**Comment by nielshulstaert** [comment]
Hi thanks you're right, I changed it because I wanted to report the same scan type as msconvert but I didn't test it thoroughly enough. I fixed it in the v1.1.7-SNAPSHOT version that I've just updated.

**Comment by timosachsenberg** [comment]
Thanks! There seem to be some new files (e.g. log file) that are not required.

**Comment by rmflight** [comment]
Yep, just tested, and the metadata seems correct.

**Comment by timosachsenberg** [comment]
I think the version string is not updated ... version="1.1.5"

**Comment by timosachsenberg** [comment]
except for the minor things like additional (unneeded? e.g. log) files in the release, no update of the version number (I guess because its a snapshot and no official version) everything seems to be fixed.
Thanks!

**Comment by rmflight** [comment]
Also, just FYI, outside of the `msnPrecursorIntensity` issue (see #47 ), the output data from both `ThermoRawFileParser` and `msconvert` appears to be identical in profile mode. Which I have to admit is awesome, because now I can just run all my conversions under Linux.

**Comment by nielshulstaert** [comment]
Hi @timosachsenberg the log file has been there since the beginning but I agree the logging to file should be optional. I've commented it out in the `log4net.config` file and updated the v1.1.7-SNAPSHOT version. I've updated the logging section in the `readme` file. 

@rmflight indeed the precursor intensity isn't there, I'll comment on that issue [#47](https://github.com/compomics/ThermoRawFileParser/issues/47). The original plan was the create a simple thermo to MGF library for linux but with the mzML output addition and figuring out the thermo native library, it took a lot longer than expected. The `msconvert` code has proven itself over the years and is a lot more mature so I'm still expecting issues to pop up. 

Thanks again for the extensive testing!
